session_subscribe_update_test.go

```diff
@@ -0,0 +1,579 @@
+		mh := NewMockControlMessageRecvQueue[*Message](ctrl)
+		
+		s := newSession(cms, mh, ProtocolQUIC, PerspectiveServer)
+		
+		// Add a local track
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		s.localTracks.confirm(42)
+		
+		// Test various priority values (0-255)
+		priorities := []uint8{0, 1, 127, 128, 255}
+		
+		for _, priority := range priorities {
+			mh.EXPECT().enqueue(context.Background(), &Message{
+				Method:             MessageSubscribeUpdate,
+				RequestID:          42,
+				TrackAlias:         100,
+				SubscriberPriority: priority,
+				StartGroup:         0,
+				StartObject:        0,
+				EndGroup:           0,
+				Forward:            1,
+			}).Return(nil)
+			
+			err := s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+				RequestID:          42,
+				StartLocation:      wire.Location{Group: 0, Object: 0},
+				EndGroup:           0,
+				SubscriberPriority: priority,
+				Forward:            1,
+				Parameters:         wire.KVPList{},
+			})
+			
+			assert.NoError(t, err, "Priority %d should be accepted", priority)
+		}
+	})
+
+	t.Run("subscribe_update_forward_flag", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		cms := NewMockControlMessageSendQueue[wire.ControlMessage](ctrl)
+		mh := NewMockControlMessageRecvQueue[*Message](ctrl)
+		
+		s := newSession(cms, mh, ProtocolQUIC, PerspectiveServer)
+		
+		// Add a local track
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		s.localTracks.confirm(42)
+		
+		// Test pause (Forward=0)
+		mh.EXPECT().enqueue(context.Background(), &Message{
+			Method:             MessageSubscribeUpdate,
+			RequestID:          42,
+			TrackAlias:         100,
+			SubscriberPriority: 128,
+			StartGroup:         0,
+			StartObject:        0,
+			EndGroup:           0,
+			Forward:            0, // Pause
+		}).Return(nil)
+		
+		err := s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 0, Object: 0},
+			EndGroup:           0,
+			SubscriberPriority: 128,
+			Forward:            0, // Pause
+			Parameters:         wire.KVPList{},
+		})
+		require.NoError(t, err)
+		
+		// Test resume (Forward=1)
+		mh.EXPECT().enqueue(context.Background(), &Message{
+			Method:             MessageSubscribeUpdate,
+			RequestID:          42,
+			TrackAlias:         100,
+			SubscriberPriority: 128,
+			StartGroup:         0,
+			StartObject:        0,
+			EndGroup:           0,
+			Forward:            1, // Resume
+		}).Return(nil)
+		
+		err = s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 0, Object: 0},
+			EndGroup:           0,
+			SubscriberPriority: 128,
+			Forward:            1, // Resume
+			Parameters:         wire.KVPList{},
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("subscribe_update_location_changes", func(t *testing.T) {
```

Comment:
We should probably test this and reject updates that widen the subscription. But I think this can be done in a single table-based testcase. The table can also include the cases for the next two functions below.